src/main/java/com/synopsys/integration/detect/tool/impactanalysis/VulnerabilityImpactAnalysisTool.java

```diff
@@ -0,0 +1,79 @@
+/**
+ * synopsys-detect
+ *
+ * Copyright (c) 2020 Synopsys, Inc.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.synopsys.integration.detect.tool.impactanalysis;
+
+import java.io.IOException;
+import java.nio.file.Path;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.synopsys.integration.detect.tool.impactanalysis.service.ImpactAnalysisService;
+import com.synopsys.integration.detect.tool.impactanalysis.service.ImpactAnalysisUploadResult;
+import com.synopsys.integration.detect.workflow.codelocation.CodeLocationNameManager;
+import com.synopsys.integration.detect.workflow.file.DirectoryManager;
+import com.synopsys.integration.exception.IntegrationException;
+import com.synopsys.integration.rest.exception.IntegrationRestException;
+import com.synopsys.integration.util.NameVersion;
+import com.synopsys.method.analyzer.core.MethodUseAnalyzer;
+
+public class VulnerabilityImpactAnalysisTool {
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    private final DirectoryManager directoryManager;
+    private final CodeLocationNameManager codeLocationNameManager;
+
+    public VulnerabilityImpactAnalysisTool(DirectoryManager directoryManager, CodeLocationNameManager codeLocationNameManager) {
+        this.directoryManager = directoryManager;
+        this.codeLocationNameManager = codeLocationNameManager;
+    }
+
+    public Path generateReport(NameVersion projectNameVersion) throws IOException {
+        MethodUseAnalyzer analyzer = new MethodUseAnalyzer();
+        String impactAnalysisCodeLocationName = codeLocationNameManager.createImpactAnalysisCodeLocationName(directoryManager.getSourceDirectory(), projectNameVersion.getName(), projectNameVersion.getVersion(), null, null);
+
+        Path outputReportPath = analyzer.analyze(directoryManager.getSourceDirectory().toPath(), directoryManager.getBinaryOutputDirectory().toPath(), impactAnalysisCodeLocationName);
```

Comment:
Should be directoryManager.getImpactOutputDirectory()

Reply:
Good catch.